src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CoseKeyHelpers.cs

```diff
@@ -0,0 +1,261 @@
+                    _ => false,
+                };
+                ;
+            }
+
+            static ECCurve MapCoseCrvToECCurve(CoseCrvId crv)
+            {
+                return crv switch
+                {
+                    CoseCrvId.P256 => ECCurve.NamedCurves.nistP256,
+                    CoseCrvId.P384 => ECCurve.NamedCurves.nistP384,
+                    CoseCrvId.P521 => ECCurve.NamedCurves.nistP521,
+                    CoseCrvId.X255519 or
+                    CoseCrvId.X448 or
+                    CoseCrvId.Ed25519 or
+                    CoseCrvId.Ed448 => throw new NotImplementedException("OKP type curves not implemented."),
+                    _ => throw new FormatException("Unrecognized COSE crv value."),
+                };
+            }
+
+            static HashAlgorithmName MapCoseKeyAlgToHashAlgorithmName(CoseKeyAlgorithm alg)
+            {
+                return alg switch
+                {
+                    CoseKeyAlgorithm.ES256 => HashAlgorithmName.SHA256,
+                    CoseKeyAlgorithm.ES384 => HashAlgorithmName.SHA384,
+                    CoseKeyAlgorithm.ES512 => HashAlgorithmName.SHA512,
+                    _ => throw new FormatException("Unrecognized COSE alg value."),
+                };
+            }
+        }
+
+        private static void WriteCoseKeyLabel(CborWriter writer, CoseKeyLabel label)
+        {
+            writer.WriteInt32((int)label);
+        }
+
+        private static CoseKeyLabel ReadCoseKeyLabel(CborReader reader, CoseKeyLabel? expectedLabel = null)
+        {
+            var label = (CoseKeyLabel)reader.ReadInt32();
+            if (label != expectedLabel)
+            {
+                throw new FormatException("Unexpected COSE key label.");
+            }
+
+            return label;
+        }
+
+        private enum CoseKeyLabel : int
+        {
+            // cf. https://tools.ietf.org/html/rfc8152#section-7.1 table 3
+            Kty = 1,
+            Kid = 2,
+            Alg = 3,
+            KeyOps = 4,
+            BaseIv = 5,
+
+            // cf. https://tools.ietf.org/html/rfc8152#section-13.1.1 table 23
+            EcCrv = -1,
+            EcX = -2,
+            EcY = -3,
+            EcD = -4,
+        };
+
+        private enum CoseCrvId : int
+        {
+            // cf. https://tools.ietf.org/html/rfc8152#section-13.1 table 22
+            P256 = 1,
+            P384 = 2,
+            P521 = 3,
+            X255519 = 4,
+            X448 = 5,
+            Ed25519 = 6,
+            Ed448 = 7,
+        }
+
+        private enum CoseKeyType : int
+        {
+            // cf. https://tools.ietf.org/html/rfc8152#section-13 table 21
+            OKP = 1,
+            EC2 = 2,
+
+            Symmetric = 4,
+            Reserved = 0,
+        }
+
+        private enum CoseKeyAlgorithm : int
+        {
+            // cf. https://tools.ietf.org/html/rfc8152#section-8.1 table 5
```

Comment:
what does cf stand for?

Reply:
Compare ... https://en.wikipedia.org/wiki/Cf.?wprov=sfti1